src/emc/motion/control.c

```diff
@@ -1316,10 +1312,8 @@ static void get_pos_cmds(long period)
 		       that fail soft limits, but we'll abort at the end of
 		       this cycle so it doesn't really matter */
 		cubicAddPoint(&(joint->cubic), joint->coarse_pos);
-		old_pos_cmd = joint->pos_cmd;
-		/* interpolate to get new one */
-		joint->pos_cmd = cubicInterpolate(&(joint->cubic), 0, 0, 0, 0);
-		joint->vel_cmd = (joint->pos_cmd - old_pos_cmd) * servo_freq;
+        /* interpolate to get new position and velocity */
+	    joint->pos_cmd = cubicInterpolate(&(joint->cubic), 0, &(joint->vel_cmd), 0, 0);
```

Comment:
minor nit: indentation level does not match surrounding code.  Please configure your editor to use 4 spaces per indentation level, and to display existing tab characters as 8 spaces.   No need to revise this PR, however.